Update porting-guides.md (#2349)
MicrosoftDocs/mixed-reality: mixed-reality-docs/porting-guides.md

## Overview

> Windows 10 includes direct support for immersive and holographic headsets. If you've built content for other devices, such as the Oculus Rift or HTC Vive, these have dependencies on libraries that exist above the operating system's platform API. Bringing existing content over to Windows Mixed Reality involves retargeting usage of these other SDKs to the Windows APIs. The [Windows platform APIs for mixed reality](https://docs.microsoft.com/uwp/api/Windows.Perception) works with Win32.

Reviewer: In a world where folks are using Unity (which abstracts the native platform APIs) or OpenXR (which replaces those Windows.Perception APIs), I'd just remove the last sentence here, and mention Win32 in the previous sentence:

```suggestion
Windows 10 includes direct support for immersive and holographic headsets. If you've built content for other devices, such as the Oculus Rift or HTC Vive, these have dependencies on libraries that exist above the operating system's platform API. Bringing existing Win32 Unity VR apps over to Windows Mixed Reality involves retargeting usage of vendor-specific VR SDKs to Unity's cross-vendor VR APIs.

```

3. Enable [Developer Mode](https://docs.microsoft.com/windows/uwp/get-started/enable-your-device-for-development)
4. Switch to the [Windows Insider Fast flights](https://blogs.technet.microsoft.com/uktechnet/2016/07/01/joining-insider-preview) through **Settings > Update & Security Section**

### Common step 3: Upgrade to the most recent build of Visual Studio

> * If you are using Visual Studio then upgrade to the most recent build

Reviewer 2: you're

Reviewer 3: ```suggestion
* If you're using Visual Studio then upgrade to the most recent build
```

* See [Install the tools](install-the-tools.md#installation-checklist) page under Visual Studio 2019

### Common step 6: Choose the correct Adapter
* In systems like notebooks with two GPUs, [target the correct adapter](rendering-in-directx.md#hybrid-graphics-pcs-and-mixed-reality-applications). This applies to Unity and native DirectX apps where a ID3D11Device is created, either explicitly or implicitly (Media Foundation), for its functionality.

## Unity porting guidance

### Unity step 1: Review the common porting steps listed above

> Review the common steps listed above to make sure your dev environment is set up correctly. In step #3, if you are using Visual Studio you should select the **Game Development with Unity** workload. You may deselect the "Unity Editor optional" component since you'll be installing a newer version of Unity in teh next step.

Reviewer 2: - Remove "dev" or update to "development"
- "you're"
- "of Unity in **the** next step."

Reviewer 3: ```suggestion
Review the common steps listed above to make sure your development environment is set up correctly. In step #3, if you're using Visual Studio you should select the **Game Development with Unity** workload. You may deselect the "Unity Editor optional" component since you'll be installing a newer version of Unity in the next step.
```

### Unity step 2: Upgrade to the latest public build of Unity with Windows MR Support
1. Download the latest [recommended public build of Unity](install-the-tools.md) with mixed reality support.
2. Save a copy of your project before you get started

> 3. Review the [documentation](https://docs.unity3d.com/Manual/UpgradeGuides.html) available from Unity on upgrading if your project was build on anolder version of Unity.

Reviewer 2: - if your project was **built**

Reviewer 3: ```suggestion
3. Review the [documentation](https://docs.unity3d.com/Manual/UpgradeGuides.html) available from Unity on upgrading if your project was built on an older version of Unity.
```

* Set architecture to "x86"
Select "Switch Platform"

> [!NOTE] 
> If your application has any dependencies on device-specific services, such as match making from Steam, you'll need to disable them at this step. You can hook up to the equivalent services that Windows provides later on.

### Unity step 5: Setup your Windows Mixed Reality hardware
1. Review steps in [Immersive headset setup](https://docs.microsoft.com/windows/mixed-reality/enthusiast-guide/before-you-start
)
2. Learn about [Using the Windows Mixed Reality simulator](using-the-windows-mixed-reality-simulator.md) and [Navigating the Windows Mixed Reality home](navigating-the-windows-mixed-reality-home.md)

> ### Unity step 7: Target your application to run on Windows Mixed Reality

Reviewer: Looks like we'll need to renumber the other sections.

1. First, you must remove or conditionally compile out any other library support specific to a particular VR SDK. Those assets frequently change settings and properties on your project in ways that are incompatible with other VR SDKs, such as Windows Mixed Reality.

>     * For example, if your project references the SteamVR SDK, you'll need to update your project to exclude those prefabs and script API calls when exporting for the Win32 build target.

Reviewer: The catch here it that targeting SteamVR already uses the Win32 build target.

The general guidance we can give here would be to use Unity's abstracted VR APIs, which handle switching across those different backing APIs for you.

Author: If you want to wordsmith that in for this I'll commit it.

Reviewer: ```suggestion
    * For example, if your project references the SteamVR SDK, you'll need to update your project to instead use Unity's common VR APIs that support both Windows Mixed Reality and SteamVR.
```
